xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -625,72 +616,85 @@ void CAESinkAUDIOTRACK::GetDelay(AEDelayStatus& status)
-      {
-        CLog::Log(LOGDEBUG,
-                  "Delay - Timestamp: {} (ms) delta: {} (ms) playtime: {} (ms) Duration: {} ms",
-                  1000.0 * (m_duration_written - playtime), delta / 1000000.0, playtime * 1000,
-                  m_duration_written * 1000);
-        CLog::Log(LOGDEBUG, "Head-Position {} Timestamp Position {} Delay-Offset: {} ms", m_headPos,
-                  m_timestampPos, 1000.0 * (m_headPos - m_timestampPos) / m_sink_sampleRate);
-      }
-      double hw_delay =
-          m_duration_written - m_timestampPos / static_cast<double>(m_sink_sampleRate);
-      // sadly we smooth the delay, so only compensate here what we did not yet smooth away
-      hw_delay -= d;
-      // sometimes at the beginning of the stream m_timestampPos is more accurate and ahead of
-      // m_headPos - don't use the computed value then and wait
-      if (hw_delay >= 0.0 && hw_delay < 1.0)
-        m_hw_delay = hw_delay;
-      if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->CanLogComponent(LOGAUDIO))
-      {
-        CLog::Log(LOGDEBUG, "HW-Delay (1): {}", hw_delay);
-      }
+    if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->CanLogComponent(LOGAUDIO))
+    {
+      CLog::Log(LOGNOTICE,
+                "Delay - Timestamp: {} (ms) delta: {} (ms) playtime: {} (ms) Duration: {} ms",
+                1000.0 * (m_duration_written - playtime), delta / 1000000.0, playtime * 1000,
+                m_duration_written * 1000);
+      CLog::Log(LOGNOTICE, "Head-Position {} Timestamp Position {} Delay-Offset: {} ms", m_headPos,
+                m_timestampPos, 1000.0 * (m_headPos - m_timestampPos) / m_sink_sampleRate);
+    }
+    double hw_delay = m_duration_written - playtime;
+    // correct by subtracting above measured delay, if lower delay gets automatically reduced
+    hw_delay -= delay;
+    // sometimes at the beginning of the stream m_timestampPos is more accurate and ahead of
+    // m_headPos - don't use the computed value then and wait
+    if (hw_delay > -1.0 && hw_delay < 1.0)
+      m_hw_delay = hw_delay;
+    else
+      m_hw_delay = 0.0;
+    if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->CanLogComponent(LOGAUDIO))
+    {
+      CLog::Log(LOGNOTICE, "HW-Delay (1): {} ms", hw_delay * 1000);
     }
   }
+
+  delay += m_hw_delay;
   if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->CanLogComponent(LOGAUDIO))
   {
-    CLog::Log(LOGDEBUG, "Delay Current: %lf", d * 1000);
+    CLog::Log(LOGNOTICE, "Combined Delay: {} ms", delay * 1000);
   }
+  if (delay < 0.0)
+    delay = 0.0;
+
+  const double d = GetMovingAverageDelay(delay);
 
+  // Audiotrack is caching more than we though it would
+  if (d > m_audiotrackbuffer_sec)
+    m_audiotrackbuffer_sec = d;
+
+  // track delay in local member
+  m_delay = d;
+  if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->CanLogComponent(LOGAUDIO))
```

Comment:
Would it not make more sense to make above:
```cpp
bool canLog = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->CanLogComponent(LOGAUDIO);
```
and then on the `if`s only the `canLog`, looks for me a bit heavy to make on stream function 5 times this big call.

Reply:
Thanks, that's a good Idea! I will do that in a follow-up PR.